Common/src/main/java/com/techarium/techarium/block/entity/multiblock/MachineCoreBlockEntity.java

```diff
@@ -30,45 +31,29 @@
 /**
  * A BlockEntity for the {@link MachineCoreBlock}.
  */
-public class MachineCoreBlockEntity extends BlockEntity implements MenuProvider {
+public class MachineCoreBlockEntity extends BlockEntity implements ExtraDataMenuProvider {
 	// We don't save the selected multiblock
 	//  1- The level isn't set when the block entity is loaded
 	//  2- I (Ketheroth) don't want to cache the resource location until I can transform it in the multiblock structure.
 
-	private Optional<MultiBlockStructure> multiblock;
+	private MultiblockStructure multiblock;
 
 	public MachineCoreBlockEntity(BlockPos pos, BlockState state) {
 		super(TechariumBlockEntities.MACHINE_CORE.get(), pos, state);
-		this.multiblock = Optional.empty();
+		this.multiblock = null;
```

Comment:
you and gravy told me to use optionals instead of null, and when I do it you remove it ? I'm sorry but there is something I don't understand here.